interview-questions.md

```diff
@@ -10,58 +10,72 @@ Then, research the question to expand on your answer. Even if you feel you have
 
   Researched answer:
-
+  "this" is in refrence to an "object" in the class method in javascript, and as per usual, I  was only half right above, an object is the cross of data and function so it gains a value and a function.
 
 
 2. What is React? Why would you use it?
 
   Your answer:
+  This I feel like I can confidently answer without much research. React is a mulitfacited open-source tool you can use to build user interfaces such as webpages and apps. It can be used to build decadent websites like "Netflix" and detailed apps like facebook and twitter. I would use React because it is one of the leading industry tools in app developement today and once you get used to the JSX language, it can be pretty straight forward to use (realize I did not say easy).  
 
   Researched answer:
+  React is a open-source JavaScript "library" used for creating web pages, apps or just UI in general that they state is "painless" to create... I suppose it depends on who you ask. React uses a virtual DOM to keep refreshing at an all time low. Instead React allows for users to interact with individual nodes that refresh when interacted with. React was released to the public less than a decade ago by the "Facebook and community" group. 
 
 
 
 3. Which lifecycle method is required in a React class component?
 
   Your answer:
+  The lifecycle method that is required to use in a React class component is "State".
 
   Researched answer:
-
+The constructor is the method required for a react class component. 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
   Your answer:
-
+  I can only recall from memory what Sarah said which is " JSX is Reacts marraige of Javascript and HTML." It has most of the functionality of Html with some Javascript capabilities. I'm not 100% on the diffrences, but I know there are limitations on how you write JavaScript into the return, for example, you need curly braces around your functions.
+  
   Researched answer:
+  One larger difference between HTML and JSX is that JSX uses className instead of class, because class is already used by javaScript. Also, JSX can only render a single component. 
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
   Your answer:
+  When you run yarn in your react, there is a large file called "node-modules" and "yarn.lock" that are nested right into your program. I'm not sure what files are "modified" when you run yarn, but I know that you are making the majority of the modifications to "src" folder and adding components to that folder, so if I had to make an educated guess, it would be that it modified the "src." 
 
   Researched answer:
-
+Ok so this was a trick question. "yarn" is a javascript packaging manager that makes up a react app, so when you run yarn you open up a browser address called "local host 3000" and you're hosting your own react website/app. 
```

Comment:
correct everything that we use in terminal with a word in the front (yarn, git, node) all reefers to a software that we are calling commands from.